app/src/main/res/values/styles.xml

```diff
@@ -459,4 +459,9 @@
     <item name="android:gravity">center_horizontal</item>
   </style>
 
+  <!-- ShapeableImageView -->
+  <style name="ShapeAppearanceOverlay.App.Circular" parent="">
```

Comment:
I'm not sure on this name--aren't all of the uses of the view supposed to be circular? How does choosing this style vs. not change the view?

Reply:
Actually initially I didn't added this style part but after building the view which should match the mock as circular was resulting as square. Hence I added this style part.

Reply:
Thanks for the image. To my original question then, why do some views not specify this style but others do? Wouldn't we want all uses of ShapeableImageView currently used by Oppia use this style since they're all CircularImageView today?

Reply:
Nice catch--what will need to be changed to bring back the shadow?

Reply:
> Nice catch--what will need to be changed to bring back the shadow?

I didn't found any solution , there is no such attribute in ShapeableImageView to add shadow.

Reply:
That's something that will need to be fixed before we can merge this PR--we don't want to regress the UI as part of this migration.

Reply:
Hey I tried searching about how we can add shadow to this view but didn't found anything.
So what shall I do now?

> That's something that will need to be fixed before we can merge this PR--we don't want to regress the UI as part of this migration.

I tried searching about how we can add shadow to this view but unfortunately nothing came up.
Now I don't know what I should do next.
Please suggest something regarding this?

Reply:
I suggest two different avenues:
1. Checking how the old library implemented its shadow and see if we can take a similar approach here
2. Looking how people generally add shadows to overlays or shaped images in Android and see if any of those approaches can be adapted here

Reply:
> I suggest two different avenues:
> 
>     1. Checking how the old library implemented its shadow and see if we can take a similar approach here
> 
>     2. Looking how people generally add shadows to overlays or shaped images in Android and see if any of those approaches can be adapted here

1. That's the main problem in CircularImageView there was an inbuilt attribute of add_shadow to add the shadow where as in ShapeableImageView there no such attribute exists.
2. I already searched it on google. It will be great help if you will suggest where I should exactly search it.

Reply:
@bhaktideshmukh out of curiosity, have you tried setting the elevation on the image? Most material components respond to elevation and show shadows for those cases.

Reply:
> @bhaktideshmukh out of curiosity, have you tried setting the elevation on the image? Most material components respond to elevation and show shadows for those cases.


Adding elevation on the image is giving view something like this

| <img src="https://user-images.githubusercontent.com/78796264/155371957-0c54be39-3021-44da-a9f8-2dde71adcda4.jpeg" width="300" />|

Here elevation is by 1 dp.

Reply:
Which view did you update the elevation for in that screenshot @bhaktideshmukh? It doesn't actually look like the shapeable image has elevation.

Reply:
After adding elevation :--

**profile_progress_header**

| Original  | Before elevation | After elevation |
|-------|---|----|
|<img src="https://user-images.githubusercontent.com/78796264/155745836-612e4665-a759-4be1-be3b-30b0dbf4508a.jpeg" width="250" />|<img src="https://user-images.githubusercontent.com/78796264/155745849-765e7f51-64e0-494c-919c-bee1b741803f.jpeg" width="250" />|<img src="https://user-images.githubusercontent.com/78796264/155745885-9ea618d3-821e-4266-88cf-2ebbcfe116a1.jpeg" width="250" />

**add_profile_activity**

| Original  | Before elevation| After elevation |
|-------|---|----|
|<img src="https://user-images.githubusercontent.com/78796264/155746523-afcc4539-fed8-4d50-9ab4-75319f08c56d.jpeg" width="250" />|<img src="https://user-images.githubusercontent.com/78796264/155746551-5fdd363d-ff9f-4d39-8c56-d4a23b9b14ab.jpeg" width="250" />|<img src="https://user-images.githubusercontent.com/78796264/155746587-eff1813f-b1eb-43b4-bc4f-2f89802bfd5e.jpeg" width="250" />

**profile_list_profile_view**

| Original  | Before elevation | After elevation |
|-------|---|----|
|<img src="https://user-images.githubusercontent.com/78796264/155746922-65e859d5-5654-481a-8df6-214347a6f558.jpeg" width="250" />|<img src="https://user-images.githubusercontent.com/78796264/155746947-e0d38507-64ab-43da-ac31-62a959503124.jpeg" width="250" />|<img src="https://user-images.githubusercontent.com/78796264/155746986-6da774e5-9478-470c-ae5d-622dad58c829.jpeg" width="250" />

Reply:
Thanks for the screenshots @bhaktideshmukh! I actually think elevation looks pretty good, and it's a really simple solution. However, I think the elevation you chose might bring on a slightly too strong shadow. Can you experimentally find an elevation value that puts the shadow visually closer to what we originally had?

Reply:
> Thanks for the screenshots @bhaktideshmukh! I actually think elevation looks pretty good, and it's a really simple solution. However, I think the elevation you chose might bring on a slightly too strong shadow. Can you experimentally find an elevation value that puts the shadow visually closer to what we originally had?

I have reduced the elevation from 20dp to 8dp and now its more closer to the shadow we had in the original one.